test/e2e/workload_test.go

```diff
@@ -0,0 +1,310 @@
+
+			By("Waiting for restore to succeed")
+			f.EventuallyRestoreSessionPhase(restoreSession.ObjectMeta).Should(Equal(v1beta1.RestoreSessionSucceeded))
+
+		})
+	})
+	Context("Backup && Restore data on different Deployment", func() {
+		BeforeEach(func() {
+			pvc = f.GetPersistentVolumeClaim()
+			err = f.CreatePersistentVolumeClaim(pvc)
+			Expect(err).NotTo(HaveOccurred())
+			deployment = f.Deployment(pvc.Name)
+
+			pvc = f.GetPersistentVolumeClaim()
+			err = f.CreatePersistentVolumeClaim(pvc)
+			Expect(err).NotTo(HaveOccurred())
+			recoveredDeployment = f.Deployment(pvc.Name)
+
+			targetref = v1beta1.TargetRef{
+				APIVersion: "apps/v1",
+				Kind:       apis.KindDeployment,
+				Name:       deployment.Name,
+			}
+		})
+		AfterEach(func() {
+			err := f.DeleteDeployment(deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			err = f.DeleteDeployment(recoveredDeployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			err = f.DeleteSecret(cred.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("Restore data on different Deployment", func() {
+			By("Creating Deployment " + deployment.Name)
+			_, err = f.CreateDeployment(deployment)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilDeploymentReady(f.KubeClient, deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating sample data inside workload")
+			err = f.CreateSampleDataInsideWorkload(deployment.ObjectMeta, apis.KindDeployment)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Reading sample data from /source/data mountPath inside workload")
+			sampleData, err := f.ReadDataFromFromWorkload(deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating storage Secret " + cred.Name)
+			err = f.CreateSecret(cred)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new repository")
+			err = f.CreateRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating BackupConfiguration" + backupCfg.Name)
+			err = f.CreateBackupConfiguration(backupCfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).Should(matcher.HaveSidecar(util.StashContainer))
+
+			By("Waiting for BackupSession")
+			f.EventuallyBackupSessionCreated(backupCfg.ObjectMeta).Should(BeTrue())
+			bs, err := f.GetBackupSession(backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check for repository status updated")
+			f.EventuallyRepository(&deployment).Should(WithTransform(f.BackupCountInRepositoriesStatus, BeNumerically(">=", 1)))
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+
+			By("Waiting to remove sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).ShouldNot(matcher.HaveSidecar(util.StashContainer))
```

Comment:
First test and this test has similarities up to this point. We can reuse this part. Check [here](https://github.com/appscode/stash/blob/7a282d91c488abc9a837c7a1a54e76970a9f3ca0/test/e2e/deployment_test.go#L76) to see how we had reused in old test.